ebib-db.el

```diff
@@ -327,6 +327,10 @@ Return t upon success, or nil if the value could not be stored."
 
 (defun ebib-db-remove-field-value (field key db)
   "Remove FIELD from entry KEY in DB."
+  (when (string= field "external")
+    (if (eq ebib-notes-storage 'one-file-per-note)
+	(delete-file (expand-file-name (ebib--create-notes-file-name key)))
+    (warn "[Ebib] `ebib-notes-storage' set to `%s', deleting not supported. Try opening the note file and deleting it manually." ebib-notes-storage)))
```

Comment:
`ebib-db.el` is the wrong place for this. `ebib-db.el` is solely for handling the struct that holds the data for a `.bib` file. Note that `ebib-db.el` doesn't depend on any other part of Ebib, so strictly speaking, the variable `ebib-notes-storage` isn't even known to it.

Deleting a notes file should happen in `ebib.el`, as a special case in `ebib-delete-field-contents`. (The `if` that checks for  the `=type=` field should probably be converted into a `cond`.)